shipyard_blueprints/snowflake/shipyard_snowflake/cli/upload.py

```diff
@@ -217,6 +217,9 @@ def main():
             f"An error occurred when attempting to upload to Snowflake: {str(e)}"
         )
         sys.exit(Database.EXIT_CODE_UNKNOWN)
+    else:
+        logger.info("Closing connection")
+        snowflake_client.close()
```

Comment:
snowflake_client.close() is a precaution but should be moved into `finally` so that it always executes

Reply:
I had it in the finally initially but put it here because in the finally clause, there is a chance the client hasn't been initialized so it would throw an exception